rilldata/rill
# feat: NOT support in alerts

## Checklist

## Summary
#### Issue addressed: 
We have had a few requests to add a NOT support in alert criteria.

#### Details:
1. Adding a `NOT` operation to our expressions.
2. Adding a `NOT` toggle per group.
3. Adding a global `NOT` toggle for the entire list of criteria. (Need design for polish)
4. Adding `NOT` indication to the alert metadata. (Need design for polish)

## Steps to Verify
1. Start cloud dev
2. Create a project with a valid dashboard
3. Create an alert
4. `NOT` toggle should show up and it should apply to the data preview in criteria tab.
5. After creating `!` should show up in the alert metadata
